java/algorithm/src/test/java/org/apache/arrow/algorithm/sort/TestIndexSorter.java

```diff
@@ -113,4 +113,45 @@ public void testSortLargeIncreasingInt() {
       }
     }
   }
+
+  @Test
+  public void testChoosePivot() {
```

Comment:
I think you need more test coverage here for selecting the midpoint from the 3 sampled points (ideally as different testes).

Reply:
Sure. I added test cases for all possible permutations for 3 values, and find bugs during the process.